src/managers/NAVManager.sol

```diff
@@ -0,0 +1,302 @@
+            assetIdToAccountId[centrifugeId][assetId] = assetAccount_;
+        }
+
+        hub.createAccount(poolId, assetAccount_, true);
+        hub.initializeHolding(
+            poolId,
+            scId,
+            assetId,
+            valuation,
+            assetAccount_,
+            equityAccount(centrifugeId),
+            gainAccount(centrifugeId),
+            lossAccount(centrifugeId)
+        );
+
+        accountCounter[centrifugeId] = index + 1;
+
+        emit InitializeHolding(scId, assetId);
+    }
+
+    /// @inheritdoc INAVManager
+    function initializeLiability(ShareClassId scId, AssetId assetId, IValuation valuation) external onlyManager {
+        uint16 centrifugeId = assetId.centrifugeId();
+        uint16 index = accountCounter[centrifugeId];
+        require(index > 0, NotInitialized());
+        require(index < type(uint16).max, ExceedsMaxAccounts());
+
+        AccountId expenseAccount_ = assetIdToAccountId[centrifugeId][assetId];
+        if (expenseAccount_.isNull()) {
+            expenseAccount_ = withCentrifugeId(centrifugeId, index);
+            assetIdToAccountId[centrifugeId][assetId] = expenseAccount_;
+        }
+
+        hub.createAccount(poolId, expenseAccount_, true);
+        hub.initializeLiability(poolId, scId, assetId, valuation, expenseAccount_, liabilityAccount(centrifugeId));
+
+        accountCounter[centrifugeId] = index + 1;
+
+        emit InitializeLiability(scId, assetId);
+    }
+
+    //----------------------------------------------------------------------------------------------
+    // Price updates
+    //----------------------------------------------------------------------------------------------
+
+    /// @inheritdoc ISnapshotHook
+    function onSync(PoolId poolId_, ShareClassId scId, uint16 centrifugeId) external {
+        require(msg.sender == address(holdings), NotAuthorized());
+        require(poolId == poolId_, InvalidPoolId());
+        _onSync(scId, centrifugeId);
+    }
+
+    /// @inheritdoc ISnapshotHook
+    function onTransfer(
+        PoolId poolId_,
+        ShareClassId scId_,
+        uint16 fromCentrifugeId,
+        uint16 toCentrifugeId,
+        uint128 sharesTransferred
+    ) external {
+        require(msg.sender == address(hub), NotAuthorized());
+        require(poolId == poolId_, InvalidPoolId());
+        require(address(navHook) != address(0), InvalidNAVHook());
+
+        navHook.onTransfer(poolId, scId_, fromCentrifugeId, toCentrifugeId, sharesTransferred);
+
+        emit Transfer(scId_, fromCentrifugeId, toCentrifugeId, sharesTransferred);
+    }
+
+    /// @inheritdoc INAVManager
+    function updateHoldingValue(ShareClassId scId, AssetId assetId) public onlyManager {
+        hub.updateHoldingValue(poolId, scId, assetId);
+        (bool isSnapshot,) = holdings.snapshot(poolId, scId, assetId.centrifugeId());
+        if (isSnapshot) _onSync(scId, assetId.centrifugeId());
+    }
```

Comment:
IIUC, `_onSync()`, will be already called by the hub inside `hub.updateHoldingValue()`, so you don't need the last two lines of this method

Reply:
it's only called in `hub.updateHoldingAmount`